### DISCUSSION TITLE
(How)does risingwave isolate both the load from mv & query?

### DISCUSSION THREAD

**Discussion by doki23** [risingwavelabs/risingwave#6716] [Q&A] [answered]
For instance, I create a materialized view of pg with a high tps and make a complex aggregative query on it. How does risingwave process the load? Do they share cpu of the same cluster component?

**Comment by yingjunwu** [comment, marked as answer]
No RisingWave does not isolate streaming workload (aka MV workload) from ad-hoc query workload. So if you have an intensive ad-hoc query workload, then stream processing will suffer, resulting in staled MV. The quick fix you may apply is to provision separate clusters - one for streaming and one for ad-hoc queries.
